python/tests/test_metadata.py

```diff
@@ -1494,6 +1511,37 @@ def test_additional_properties(self):
         ):
             metadata.MetadataSchema(schema)
 
+    def test_unrequired_property_needs_default(self):
+        schema = {
+            "codec": "struct",
+            "type": "object",
+            "properties": {
+                "int": {"type": "number", "binaryFormat": "i"},
+                "float": {"type": "number", "binaryFormat": "d"},
+            },
+            "required": ["float"],
+        }
+        with pytest.raises(
+            exceptions.MetadataSchemaValidationError,
+            match="Optional property 'int' must have a default value",
+        ):
+            self.encode(schema, {"float": 5.5})
```

Comment:
This should be raised when we create the schema rather than when encoding data I think? (probably is, but the test should reflect this)

Reply:
Modified test to just create schema, the schema was where the error was already being rasied.